lib/ember-orbit/schema.js

```diff
@@ -151,13 +151,13 @@ var Schema = Ember.Object.extend({
           var rel = record.__rel[link] = {};
           linkValue.forEach(function(id) {
             if (typeof id === 'object') {
-              id = id.primaryId;
+              id = id.get('primaryId');
```

Comment:
Please change to `get(id, 'primaryId')` (`get` is defined as `Ember.get` above).